### PR TITLE
Adopt micorbundle and use esmodule by default

### PR DESCRIPTION
**Pull request by huozhi**
* rm "main" field and cjs bundle
* add `"type": "module"`
* couple of fixes to make tests work again & rebase master

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   "license": "MIT",
   "scripts": {
     "build": "microbundle --no-sourcemap -f modern,es,cjs",
+    "postbuild": "echo \";Object.defineProperty(exports,'__esModule',{value:true})\" >> dist/index.js",
```
Comment (huozhi) [inline package.json:19]: @shuding seems to work (checking the SSR sandbox demo), but kinda a dirty workaround..

**Comment by shuding** [comment]
Is CJS removed in this PR? (it's a bit hard to read the diff, make sure you have rebased correctly)

**Comment by huozhi** [comment]
> Is CJS removed in this PR? (it's a bit hard to read the diff, make sure you have rebased correctly)

@shuding  have rebased master and re-target the dist branch to master branch, easier to go through the changes now

**Comment by shuding** [comment]
https://github.com/vercel/swr/pull/881#issuecomment-789711933

**Pull request closed without merging**